packages/node/src/integrations/tracing/nest/sentry-nest-instrumentation.ts

```diff
@@ -103,21 +110,16 @@ export class SentryNestInstrumentation extends InstrumentationBase {
                 const [req, res, next, ...args] = argsUse;
                 const prevSpan = getActiveSpan();
 
-                return startSpanManual(getMiddlewareSpanOptions(target), (span: Span) => {
-                  const nextProxy = new Proxy(next, {
-                    apply: (originalNext, thisArgNext, argsNext) => {
-                      span.end();
-
-                      if (prevSpan) {
-                        return withActiveSpan(prevSpan, () => {
-                          return Reflect.apply(originalNext, thisArgNext, argsNext);
-                        });
-                      } else {
-                        return Reflect.apply(originalNext, thisArgNext, argsNext);
-                      }
-                    },
-                  });
+                // Check that we can reasonably assume that the target is a middleware.
+                // Without this guard, instrumentation will fail if a function named 'use' on a service, which is
+                // decorated with @Injectable, is called.
+                if (!(next satisfies NextFunction)) {
```

Comment:
Just fyi, satisfies is a type annotation and has no effect during runtime. What was your intention with this if statement?

Reply:
To early-return before we try to proxy next. Thanks for the hint, interesting that it does seem to work nevertheless.

I added tests first to verify the behavior, then introduced this check and that fixed it